metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/ParentChildRelMetadataService.java

```diff
@@ -0,0 +1,78 @@
+package com.netflix.metacat.common.server.usermetadata;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.server.model.ChildInfo;
+import com.netflix.metacat.common.server.model.ParentInfo;
+
+import java.util.Set;
+
+/**
+ * Parent-Child Relationship Metadata Service  API.
+ *
+ * @author yingjianw
+ */
+public interface ParentChildRelMetadataService {
+    /**
+     * Establishes a parent-child relationship with a specified relation type.
+     * Currently, exceptions are thrown in the following cases:
+     * 1. Attempting to create a child table as the parent of another child table.
+     * 2. A child table having more than one parent.
+     *
+     * @param parentName    the name of the parent entity
+     * @param childName     the name of the child entity
+     * @param relationType  the type of the relationship
+     */
+    void createParentChildRelation(
+        QualifiedName parentName,
+        QualifiedName childName,
+        String relationType
+    );
+
+    /**
+     * Renames `oldName` to `newName` in the parentChildRelationship store.
+     * This involves two steps:
+     * 1. Rename all records where the child is `oldName` to `newName`.
+     * 2. Rename all records where the parent is `oldName` to `newName`.
+     * Since a child cannot be a parent of another entity, only one of these actions will need to be performed.
+     *
+     * @param oldName  the current name to be renamed
+     * @param newName  the new name to rename to
+     */
+    void rename(
+        QualifiedName oldName,
+        QualifiedName newName
+    );
+
+    /**
+     * Removes the entity from the parentChildRelationship store.
+     * An exception is thrown if an attempt is made to drop a parent table that still has existing child tables.
+     * Note that only dropping a child table will result in the physical deletion of records.
+     * When all children of a parent table are dropped, no records remain.
+     * Consequently, at then time when dropping the parent table, there should not have any database records
+     * with a parent reference to the dropping parent table.
+     *
+     * @param name  the name of the entity to drop
+     */
+    void drop(
+        QualifiedName name
+    );
+
+    /**
+     * get the parent for the input name.
+     *
+     * @param name  name
+     * @return parentInfo
+     */
+    ParentInfo getParent(
```

Comment:
Should we return a Set of ParentInfo even though we only have one parent right now?

Downstream services that depends on this field can implement with the assumption that the parent filed will always be a set and they don't need to worry about changing their implementations once we support multi-parents